dbt_integration.py

```diff
@@ -558,6 +571,38 @@ def get_columns(self, node: "ManifestNode") -> List["ColumnInfo"]:
             columns.extend(result.table.column_names)
         return columns
 
+    def get_catalog(self) -> Dict[str, Any]:
+        """Get catalog from adapter"""
+        catalog_table: agate.Table = agate.Table([])
+        catalog_data: List[PrimitiveDict] = []
+        exceptions: List[Exception] = []
+        logger.debug("getting connection")
+        try:
+            with self.adapter.connection_named("generate_catalog"):
+                logger.debug("Generating catalog...")
+                catalog_table, exceptions = self.adapter.get_catalog(self.dbt)
```

Comment:
What about a model that refers to another database?

Reply:
if docs generate works, this should work too. adapter's get_catalog is what is called for docs generate.